pkg/acquisition/modules/docker/docker.go

```diff
@@ -872,15 +909,90 @@ func ReadTailScanner(scanner *bufio.Scanner, out chan string, t *tomb.Tomb) erro
 	return scanner.Err()
 }
 
+// isContainerStillRunning checks if a container is still running via Docker API
+func (d *DockerSource) isContainerStillRunning(ctx context.Context, container *ContainerConfig) bool {
+	containerInfo, err := d.Client.ContainerInspect(ctx, container.ID)
+	if err != nil {
+		if errdefs.IsNotFound(err) {
+			// Container does not exist - it was removed
+			container.logger.Debugf("container %s not found, was removed", container.Name)
+			return false
+		}
+		// Other errors (connection issues, etc.) - assume container is still running
+		container.logger.Debugf("failed to inspect container %s: %v, assuming still running", container.Name, err)
+		return true
+	}
+
+	isRunning := containerInfo.State.Running
+	container.logger.Debugf("container %s running status: %v", container.Name, isRunning)
+	return isRunning
+}
+
+// isServiceStillRunning checks if a service still exists via Docker API
+func (d *DockerSource) isServiceStillRunning(ctx context.Context, service *ContainerConfig) bool {
+	_, _, err := d.Client.ServiceInspectWithRaw(ctx, service.ID, dockerTypes.ServiceInspectOptions{})
+	if err != nil {
+		if errdefs.IsNotFound(err) {
+			// Service does not exist - it was removed
+			service.logger.Debugf("service %s not found, was removed", service.Name)
+			return false
+		}
+		// Other errors (connection issues, etc.) - assume service still exists
+		service.logger.Debugf("failed to inspect service %s: %v, assuming still running", service.Name, err)
+		return true
+	}
+
+	service.logger.Debugf("service %s still exists", service.Name)
+	return true
+}
+
 func (d *DockerSource) TailContainer(ctx context.Context, container *ContainerConfig, outChan chan types.Event, deleteChan chan *ContainerConfig) error {
-	container.logger.Info("start tail")
+	container.logger.Info("start monitoring")
+
+	for {
+		select {
+		case <-container.t.Dying():
+			container.logger.Infof("tail stopped for container %s", container.Name)
+			return nil
+		default:
+		}
+
+		err := d.tailContainerAttempt(ctx, container, outChan)
+
+		if err == nil {
+			// Successful completion - container was stopped gracefully
+			return nil
+		}
+
+		// Check container health to determine if we should retry or give up
+		containerHealthy := d.isContainerStillRunning(ctx, container)
+
+		if !containerHealthy {
+			// Container is dead/stopped - don't retry, remove from monitoring
+			container.logger.Infof("container %s is no longer running, removing from monitoring: %v", container.Name, err)
+			deleteChan <- container
```

Comment:
im un decided if the container should self delete, since we get docker events now we already get this.

my fear is that it may cause a race condition like:

docker event die -> docker event alive -> container self delete

Reply:
"we get docker events" -- if the proxy was down during the container death, we are not notified of that and remove it when we can confirm it's missing, right?